tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs

```diff
@@ -394,17 +394,26 @@ public void AddSanitizerToRecording(string recordingId, RecordedTestSanitizer sa
         {
             if (PlaybackSessions.TryGetValue(recordingId, out var playbackSession))
             {
-                playbackSession.AdditionalSanitizers.Add(sanitizer);
+                lock (playbackSession)
```

Comment:
@HarshaNalluru Ubuntu is fast enough that I think we're returning `200 ok` before the actual addition of each sanitizer is _completed_.  By locking this, this should prevent the sanitization pass from even starting until the lock has been released. 

Famous last words :)